### ISSUE TITLE
MatFormField should document error accessibility best practices

### ISSUE THREAD

**Issue by jelbourn** [angular/components#22631]
Right now the only difference between text shown in `<mat-hint>` and in `<mat-error>` is the color. This is an accessibility issue because it may not be clear that a message being shown is marked as an error message. 

I think that adding a marker directly to `<mat-error>` would be too opinionated for Angular Material, but we should document a best practice saying that applications should add some sort of marker, such as text or an icon, indicating that the contents of `<mat-error>` represent an error message.

**Comment by umairhafeez-r** [comment]
@jelbourn According to the following documentation, using `role="alert"` should be sufficient, and that already exists on `mat-error` directive.

https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA19.html

However, I see a different problem here. When the form field has error, the underlying `input` element still has `aria-invalid="false"`. If it gets proper value, the screen readers will read it out as `invalid`.

I ran a small test with VoiceOver on my MacBook and here is the video. You can see how when I changed the aria-invalid from false to true, the VoiceOver reads it as `required invalid`.

<a href="https://www.loom.com/share/2d9a1d3ac2214aad9e462dd23504020e">
    <p>Form field | Angular Material - Watch Video</p>
    <img style="max-width:300px;" src="https://cdn.loom.com/sessions/thumbnails/2d9a1d3ac2214aad9e462dd23504020e-with-play.gif">
  </a>

**Comment by umairhm** [comment]
@jelbourn According to the following documentation, using `role="alert"` should be sufficient, and that already exists on `mat-error` directive.

https://www.w3.org/WAI/WCAG21/Techniques/aria/ARIA19.html

However, I see a different problem here. When the form field has error, the underlying `input` element still has `aria-invalid="false"`. If it gets proper value, the screen readers will read it out as `invalid`.

I ran a small test with VoiceOver on my MacBook and here is the video. You can see how when I changed the aria-invalid from false to true, the VoiceOver reads it as `required invalid`.

<a href="https://www.loom.com/share/2d9a1d3ac2214aad9e462dd23504020e">
    <p>Form field | Angular Material - Watch Video</p>
    <img style="max-width:300px;" src="https://cdn.loom.com/sessions/thumbnails/2d9a1d3ac2214aad9e462dd23504020e-with-play.gif">
  </a>

**Comment by umairhm** [comment]
Gotcha! So by marker, you mean some icon that is commonly used for error states or maybe putting the word "Error" in the content itself, so the users can actually figure out that it's an error even if they don't recognize the red color. Is this correct?
